dask_expr/_expr.py

```diff
@@ -2758,24 +2767,37 @@ def optimize(expr: Expr, fuse: bool = True) -> Expr:
         Input expression to optimize
     fuse:
         whether or not to turn on blockwise fusion
+    common_subplan_elimination : bool
```

Comment:
If I'm not mistaken, our default is actually a very radical form of "common subplan elimination". I think this feature is actually quite the opposite and rather something like "replicate common subplan"? I guess other engines are doing it the other way round, aren't they?

Reply:
I took the name from here: https://docs.pola.rs/py-polars/html/reference/lazyframe/api/polars.LazyFrame.explain.html

But I don't have strong feelings either way

Reply:
There is also https://en.wikipedia.org/wiki/Common_subexpression_elimination with a simple example =)

Reply:
Naming is hard. @hendrikmakait any thoughts?

Reply:
Sorry that I'm late to the party. `dask-expr` approaches CSE from the other end, defaulting to as much CSE as possible. This makes it less intuitive. As mentioned in #896, `common_subplan_elimination=False` does not mean that we don't do any CSE but instead that we try to reduce it. I'm fine with the argument name for now that you've cleaned it up, let's see if this leads to some confusion in the future.

Reply:
Regarding Common Sub**expression** Elimination vs Common Sub**plan** Elimination: I think the latter is more common for data systems and compilers/IR-based systems.

Reply:
Interestingly. `polars` seems to have both sub**expression** (https://github.com/pola-rs/polars/blob/128803b237dc13d0522c22dbccae1257ae30477e/crates/polars-plan/src/logical_plan/optimizer/cse_expr.rs) as well as sub**plan** elimination (https://github.com/pola-rs/polars/blob/128803b237dc13d0522c22dbccae1257ae30477e/crates/polars-plan/src/logical_plan/optimizer/cse.rs).

I'm not _entirely_ sure what the difference is, but from what I understand `cse_expr` deals with eliminating duplication within a single expressions (`a.sum() + b.sum() + a.sum()`) and `cse` deals with what eliminating subgraphs across the entire graph.

That is, `cse_expr` corresponds to _local_ CSE and `cse` to _global_ CSE in the link Florian shared.